app/src/main/java/com/xmartlabs/scasas/criminalintent/ui/crime/simple/CrimePagerActivity.java

```diff
@@ -0,0 +1,69 @@
+package com.xmartlabs.scasas.criminalintent.ui.crime.simple;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.view.ViewPager;
+
+import com.annimon.stream.Objects;
+import com.annimon.stream.Stream;
+
+import com.xmartlabs.scasas.criminalintent.R;
+import com.xmartlabs.scasas.criminalintent.controller.CrimeController;
+import com.xmartlabs.scasas.criminalintent.model.Crime;
+
+import java.util.List;
+import java.util.UUID;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by scasas on 2/14/17.
+ */
+public class CrimePagerActivity extends FragmentActivity {
+  private static final String EXTRA_CRIME_ID = CrimePagerActivity.class.getCanonicalName() + ".crime_id";
+
+  @BindView(R.id.activity_crime_pager_view_pager)
+  ViewPager viewPager;
+
+  private final List<Crime> crimes = CrimeController.getInstance().getCrimes();
+
+  public static Intent newIntent(Context packageContext, UUID crimeId) {
+    return new Intent(packageContext, CrimePagerActivity.class)
+        .putExtra(EXTRA_CRIME_ID, crimeId);
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setupLayout();
+    setupCrimePagerAdapter();
+    UUID crimeId = obtainCrimeIdFromCall();
+    setCurrentItem(crimeId);
+  }
+
+  private void setCurrentItem(UUID crimeId) {
+    Stream.range(0, crimes.size() - 1)
+        .filter(index -> Objects.equals(crimes.get(index).getId(), crimeId))
```

Comment:
because he need the index. BTW: the index is wrong, should be `(0, crimes.size())`